### PR TITLE
adding 2.15 release notes

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description

2.15 release notes

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Version 2.15.0.0 Release Notes
+
+Compatible with OpenSearch Dashboards 2.15.0
+
+### Enhancements
+
+* support MDS in feature anywhere ([#767](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/767))
+* update create detector page and detector detail page to add custom result index lifecycle management settings ([#770](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/770))
+* Update Frontend for Custom Result Index Query and Fix Issues ([#772](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/772))
+* make custom result index lifecycle management in AD optional ([#777](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/777))
```
Comment (jackiehanyang) [inline release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md:10]: we can remove this, or combine it with the second item (line 8)

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+### Bug Fixes
+
+* Fix handling of special characters in categorical values ([#757](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/757))
+* Fix Warning Message About Custom Result Index Despite Existing Indices ([#759](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/759))
+* fix a MDS related bug on feature anywhere page ([#773](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/773))
```
Comment (jackiehanyang) [inline release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md:16]: we can remove this, it's a bug fix for the line 7 item

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+* mds feature anywhere associated detectors fix ([#778](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/778))
```
Comment (jackiehanyang) [inline release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md:17]: same as above, please remove this one

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+## Version 2.15.0.0 Release Notes
+
+Compatible with OpenSearch Dashboards 2.15.0
+
+### Enhancements
```
Comment (jackiehanyang) [inline release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md:5]: let's call it features?

Comment (amitgalitz) [reply release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md:5]: all 3 or just the first 2?

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md (hunk 1 above)
Comment (amitgalitz) [reply release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md]: removed

File: release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md (hunk 2 above)
Comment (amitgalitz) [reply release-notes/opensearch-anomaly-detection-dashboards.release-notes-2.15.0.0.md]: removed

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**